pkg/metrics/metrics.go

```diff
@@ -50,7 +53,10 @@ var (
 		[]string{
 			ReasonLabel,
 			NodePoolLabel,
+			ZoneLabel,
+			ArchLabel,
 			CapacityTypeLabel,
+			InstanceTypeLabel,
```

Comment:
InstanceType label will have a fair bit of cardinality. Azure has 800+ Instance types that karpenter can end up selecting so this would significantly increase the cardinality of these metric series